File: frontend/src/services/customerService.ts
import { apiClient } from '@/config/apiClient';
import type { ApiResponse, PageResponse, TableParams } from '@/types';

export interface CustomerStatsResponse {
  id: number;
  name: string;
  phone: string;
  isActive: boolean;
  totalOrders: number;
  totalSpent: number;
}

export interface CustomerResponse {
  id: number;
  name: string;
  phone: string;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CustomerRequest {
  name: string;
  phone: string;
}

export const customerService = {
  getAll: (params?: TableParams): Promise<ApiResponse<PageResponse<CustomerStatsResponse>>> => {
    return apiClient.get('/customers', { params });
  },

  search: (query: string): Promise<ApiResponse<CustomerResponse[]>> => {
    return apiClient.get('/customers/search', { params: { query } });
  },

  getById: (id: number): Promise<ApiResponse<CustomerResponse>> => {
    return apiClient.get(`/customers/${id}`);
  },

  create: (data: CustomerRequest): Promise<ApiResponse<CustomerResponse>> => {
    return apiClient.post('/customers', data);
  },

  update: (id: number, data: CustomerRequest): Promise<ApiResponse<CustomerResponse>> => {
    return apiClient.put(`/customers/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/customers/${id}`);
  },

  getOrders: (id: number): Promise<ApiResponse<any[]>> => {
    return apiClient.get(`/customers/${id}/orders`);
  },
};

File: backend/src/main/java/com/bubbleflow/service/CustomerService.java
package com.bubbleflow.service;

import com.bubbleflow.dto.customer.request.CustomerRequest;
import com.bubbleflow.dto.customer.response.CustomerResponse;
import com.bubbleflow.dto.customer.response.CustomerStatsResponse;
import com.bubbleflow.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface CustomerService {
    PageResponse<CustomerStatsResponse> getAll(String search, Pageable pageable);
    List<CustomerResponse> search(String query);
    CustomerResponse getById(Long id);
    CustomerResponse create(CustomerRequest request);
    CustomerResponse update(Long id, CustomerRequest request);
    void delete(Long id);
    List<com.bubbleflow.dto.order.response.OrderResponse> getOrdersByCustomerId(Long customerId);
}
